src/dfx/src/commands/deps/deploy.rs

```diff
@@ -0,0 +1,121 @@
+use crate::lib::deps::{
+    get_canister_prompt, get_pull_canister_or_principal, get_pull_canisters_in_config,
+    get_pulled_wasm_path, load_init_json, load_pulled_json, validate_pulled, InitJson,
+};
+use crate::lib::environment::Environment;
+use crate::lib::error::DfxResult;
+use crate::lib::root_key::fetch_root_key_if_needed;
+use crate::lib::state_tree::canister_info::read_state_tree_canister_controllers;
+
+use anyhow::{anyhow, bail};
+use candid::Principal;
+use clap::Parser;
+use fn_error_context::context;
+use ic_agent::Agent;
+use ic_utils::interfaces::{management_canister::builders::InstallMode, ManagementCanister};
+use slog::{info, Logger};
+
+/// Deploy pulled dependencies.
+#[derive(Parser)]
+pub struct DepsDeployOpts {
+    /// Specify the canister to deploy. You can specify its name (as defined in dfx.json) or Principal.
+    /// If not specified, all pulled canisters will be deployed.
+    canister: Option<String>,
+}
+
+pub async fn exec(env: &dyn Environment, opts: DepsDeployOpts) -> DfxResult {
+    let logger = env.get_logger();
+    let pull_canisters_in_config = get_pull_canisters_in_config(env)?;
+    if pull_canisters_in_config.is_empty() {
+        info!(logger, "There are no pull dependencies defined in dfx.json");
+        return Ok(());
+    }
+
+    let project_root = env.get_config_or_anyhow()?.get_project_root().to_path_buf();
+    let pulled_json = load_pulled_json(&project_root)?;
+    validate_pulled(&pulled_json, &pull_canisters_in_config)?;
+
+    let init_json = load_init_json(&project_root)?;
+
+    fetch_root_key_if_needed(env).await?;
+    let agent = env
+        .get_agent()
+        .ok_or_else(|| anyhow!("Cannot get HTTP client from environment."))?;
+
+    let canister_ids = match &opts.canister {
+        Some(canister) => {
+            let canister_id = get_pull_canister_or_principal(canister, &pull_canisters_in_config)?;
+            vec![canister_id]
+        }
+        None => pulled_json.canisters.keys().copied().collect(),
+    };
+    for canister_id in canister_ids {
+        // Safe to unwrap:
+        // caniseter_ids are guranteed to exist in pulled.json
+        let pulled_canister = pulled_json.canisters.get(&canister_id).unwrap();
```

Comment:
It turns out this unwrap isn't safe.

Repro steps:
1. Set up a pullable canister
2. `dfx deps pull && dfx deps init`
3. `dfx deps deploy <some other canister id>`

I used this metadata, for example:
```
      "metadata": [
          {
              "name": "dfx:wasm_url",
              "content": "https://github.com/dfinity/sdk/raw/master/e2e/assets/wasm/identity/main.wasm"
          },
          {
              "name": "dfx:wasm_hash",
              "content": "bd27eb59219b5a810621d577c6d8a56ea68529fa53bbd855fdb8bd9ea328f2c3"
          },
          {
              "name": "dfx:deps",
              "content": ""
          }
      ]
```

Outcome:
```
$ dfx-r -v -v deps deploy zrrcc-3aaaa-aaaai-abzkq-cai
Trace mode enabled. Lots of logs coming up.
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', src/dfx/src/commands/deps/deploy.rs:55:71
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```